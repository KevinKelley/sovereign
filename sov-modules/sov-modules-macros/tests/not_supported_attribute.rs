use sov_modules_api::Context;
use sov_modules_macros::ModuleInfo;
use sov_state::StateMap;

#[derive(ModuleInfo)]
struct TestStruct<C: Context> {
    #[other]
    test_state1: StateMap<u32, String, C::Storage>,

    #[state]
    test_state2: StateMap<C::PublicKey, String, C::Storage>,
}

fn main() {}
