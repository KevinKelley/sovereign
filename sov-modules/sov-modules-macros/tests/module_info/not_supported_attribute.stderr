error: Only `#[module]` or `#[state]` attributes are supported.
 --> tests/module_info/not_supported_attribute.rs:8:5
  |
8 |     test_state1: StateMap<u32, String, C::Storage>,
  |     ^^^^^^^^^^^

error: cannot find attribute `other` in this scope
 --> tests/module_info/not_supported_attribute.rs:7:7
  |
7 |     #[other]
  |       ^^^^^
